tests/middleware/test_wsgi.py

```diff
@@ -1,17 +1,38 @@
 import io
 import sys
-from typing import TYPE_CHECKING, AsyncGenerator, List
+from importlib import reload
+from typing import TYPE_CHECKING, AsyncGenerator, Awaitable, Callable, List
+from unittest import mock
 
 import httpx
 import pytest
 
 from uvicorn._types import Environ, StartResponse
-from uvicorn.middleware.wsgi import WSGIMiddleware, build_environ
+from uvicorn.middleware import wsgi
 
 if TYPE_CHECKING:
     from asgiref.typing import HTTPRequestEvent, HTTPScope
 
 
+def environ_switcher(
+    async_test: Callable[[], Awaitable[None]]
+) -> Callable[[bool], Awaitable[None]]:
+    from uvicorn.middleware import wsgi
+
+    async def test_wrapper(use_a2wsgi: bool) -> None:
+        if use_a2wsgi:
+            await async_test()
+        else:
+            with mock.patch.dict(sys.modules, {"a2wsgi": None}):
+                reload(wsgi)
+
+                await async_test()
+
+            reload(wsgi)
+
+    return test_wrapper
+
+
```

Comment:
Now we do this, and avoid the `@environ_switcher` magic, and the individual parametrize on each test.

```suggestion
@pytest.fixtures(param=wsgi._WSGIMiddleware, a2wsgi.WSGIMiddleware)
def wsgi_middleware_class(request):
    return request.param
```

Reply:
Done, much better now :smile: